src/features/blockchain/utils/wallet.ts

```diff
@@ -0,0 +1,64 @@
+import { WalletStore, WalletTypes } from '../stores/walletStore';
+import { ChainType } from '../types/chain';
+import { chainRegistry } from '../config/chainsRegistry';
+
+/**
+ * Detects if the current device is a mobile device based on user agent and viewport width
+ * @returns boolean - true if the device is mobile, false otherwise
+ */
+export function isMobileDevice(): boolean {
+  //  Use regex to detect mobile
+  const isMobileUserAgent =
+    /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
+      window.navigator.userAgent,
+    );
+
+  const isMobileViewport = window.innerWidth <= 768;
+
+  //  But also confirm the viewport (chrome devtools throws a false positive based on agent-alone)
+  return isMobileUserAgent && isMobileViewport;
+}
```

Comment:
This is the main update here - don't just check the `userAgent`

Reply:
I thikn we should move this viewport logic to the core space since it's not specific to the blockchain logic, just currently being used there.  I could also see a follow on ticket here to integrate this into the app itself.